### PR TITLE
JENKINS-67109 Fix maven installations

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-67109](https://issues.jenkins-ci.org/browse/JENKINS-67109).

### Proposed changelog entries

* Maven tool configuration works again (regression in 2.320).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@janfaracik 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/lib/form/entry.jelly
Comment (timja) [inline core/src/main/resources/lib/form/entry.jelly:75]: @janfaracik is it ok to restore this?

I haven't been able to track down what's missing, some notes in https://issues.jenkins.io/browse/JENKINS-67109

**Review by basil — APPROVED** [review]

File: core/src/main/resources/lib/form/entry.jelly
Comment (janfaracik) [reply core/src/main/resources/lib/form/entry.jelly:75]: Looks good to me 👍

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by jglick** [comment]
> regression in 2.320

For code archaeology I would suggest

* linking here to the PR introducing the regression
* linking this Jira issue to the issue associated with the earlier change, if any

**Pull request merged**
